engine/classes/Elgg/Application.php

```diff
@@ -0,0 +1,293 @@
+<?php
+
+namespace Elgg;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * @access private
+ *
+ * @package    Elgg.Core
+ * @subpackage Application
+ * @since      2.0.0
+ */
+class Application {
+
+	/**
+	 * @var \Elgg\Config
+	 */
+	private $config;
+
+	/**
+	 * @var string
+	 */
+	private $engineDir;
+
+	/**
+	 * @var string
+	 */
+	private $wwwRoot;
```

Comment:
Ah, I'm mixing up PRs, but still. We should distinguish between the root of elgg and the root of the app since those may not be the same going forward...